Test flake on TestDedupeRepos

Reproducible steps - 
1. clone main
2. cd `pkg/downloads` 
3. `go test -run TestDedupeRepos -count=1` <- this should pass
4. `go test -run TestDedupeRepos -count=100` <- this fails pass

Seeing this flake on PR #13617, I have not looked into the cause at all

```go
--- FAIL: TestDedupeRepos (0.00s)
    --- FAIL: TestDedupeRepos/no_duplicates (0.00s)
        manager_test.go:661: received:
            [{"name":"","url":"https://example.com/charts2","username":"","password":"","certFile":"","keyFile":"","caFile":"","insecure_skip_tls_verify":false,"pass_credentials_all":false} {"name":"","url":"https://example.com/charts","username":"","password":"","certFile":"","keyFile":"","caFile":"","insecure_skip_tls_verify":false,"pass_credentials_all":false}]
            want:
            [{"name":"","url":"https://example.com/charts","username":"","password":"","certFile":"","keyFile":"","caFile":"","insecure_skip_tls_verify":false,"pass_credentials_all":false} {"name":"","url":"https://example.com/charts2","username":"","password":"","certFile":"","keyFile":"","caFile":"","insecure_skip_tls_verify":false,"pass_credentials_all":false}]
---... happens more...---
--- FAIL: TestDedupeRepos (0.00s)
    --- FAIL: TestDedupeRepos/no_duplicates (0.00s)
        manager_test.go:661: received:
           <same-output-again>
FAIL
exit status 1
```

--- a/pkg/downloader/manager_test.go
+++ b/pkg/downloader/manager_test.go
@@ -22,6 +22,8 @@ import (
 	"reflect"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+
 	"helm.sh/helm/v4/pkg/chart"
 	"helm.sh/helm/v4/pkg/chart/loader"
 	chartutil "helm.sh/helm/v4/pkg/chart/util"
@@ -657,9 +659,8 @@ func TestDedupeRepos(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := dedupeRepos(tt.repos); !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("received:\n%v\nwant:\n%v", got, tt.want)
-			}
+			got := dedupeRepos(tt.repos)
+			assert.ElementsMatch(t, tt.want, got)
 		})
 	}
 }

